pkg/forward/forward.go

```diff
@@ -236,7 +236,7 @@ func (isdf *InterStepDataForward) forwardAChunk(ctx context.Context) {
 			return
 		}
```

Comment:
So, my understanding is that we are getting the smallest watermark from all publishers and using that as our Idle Watermark. Why do we restrict it to just the current partition?